internal/controllers/common_controller.go

```diff
@@ -98,3 +101,26 @@ func setResolutionStatus(ext *ocv1alpha1.ClusterExtension, resStatus *ocv1alpha1
 func setInstallStatus(ext *ocv1alpha1.ClusterExtension, installStatus *ocv1alpha1.ClusterExtensionInstallStatus) {
 	ext.Status.Install = installStatus
 }
+
+func setStatusProgressing(ext *ocv1alpha1.ClusterExtension, err error) {
+	progressingCond := metav1.Condition{
+		Type:               ocv1alpha1.TypeProgressing,
+		Status:             metav1.ConditionFalse,
+		Reason:             ocv1alpha1.ReasonSuccess,
+		Message:            "desired state reached",
+		ObservedGeneration: ext.GetGeneration(),
+	}
+
+	if err != nil {
+		progressingCond.Status = metav1.ConditionTrue
+		progressingCond.Reason = ocv1alpha1.ReasonRetrying
+		progressingCond.Message = err.Error()
```

Comment:
Should we go ahead and add the name, version, and resolved image reference of the bundle here?

That way, the PR that removes the `status.resolved` field can literally just remove the field and not have to remember to come here and update the message.

Reply:
I would expect the string we receive from the `err.Error()` call to return a string that contains this information + the error encountered. I can update the errors we send to this `setStatusProgressing` error to have that information.